clab/export.go

```diff
@@ -41,10 +42,18 @@ type TopologyExport struct {
 	NodeConfigs map[string]*types.NodeConfig `json:"nodeconfigs,omitempty"`
 }
 
+//go:embed export_templates/auto.tmpl
+var defaultExportTemplate string
+
 // exportTopologyDataWithTemplate generates and writes topology data file to w using a template.
 func (c *CLab) exportTopologyDataWithTemplate(_ context.Context, w io.Writer, p string) error {
-	n := filepath.Base(p)
-	t, err := template.New(n).
+	var n string
+	if len(p) != 0 {
+		n = filepath.Base(p)
+	} else {
+		n = "auto.tmpl"
+	}
```

Comment:
I think, in this case, the template name can be unconditionally named "export" without having the if/else branching.